### PR TITLE
Asynchronous transitions - part 1

### PR DESCRIPTION
**Pull request by mkurek**
- added possibility to run transition on rq worker
- added views with preview of job status

Notice that async method (method executed on worker) requires DB access to update Job status and progress.

### CODE REVIEW

File: src/ralph/lib/transitions/async.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            with transaction.atomic():
+                func(instances=[obj], **defaults)
+        except RescheduleAsyncTransitionActionLater as e:
+            # action is not ready - reschedule this job later and continue
+            # when you left off
+            transition_job.reschedule()
+            return
+        except Exception as e:
+            logger.exception(e)
+            tja.status = TransitionJobActionStatus.FAILED
```
Comment (ar4s) [inline src/ralph/lib/transitions/async.py:124]: if action failed then job fail

File: src/ralph/lib/transitions/async.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    """
+    started_actions = 0
+    for action in executed_actions:
+        if action.status == TransitionJobActionStatus.FAILED:
+            logger.warning(
+                'Action {} has failed. Marked job {} as failed too'.format(
+                    action, job
+                )
+            )
+            job.status = JobStatus.FAILED
```
Comment (szok) [inline src/ralph/lib/transitions/async.py]: job.fail() ?

File: src/ralph/lib/transitions/async.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    try:
+        _check_previous_actions(transition_job, executed_actions)
+    except AsyncTransitionError:
+        return
+
+    completed_actions_names = set([
+        tja.action_name for tja in executed_actions
+        if tja.status != TransitionJobActionStatus.STARTED
+    ])
+    for action in _order_actions_by_requirements(transition.actions.all(), obj):
```
Comment (mkurek) [inline src/ralph/lib/transitions/async.py]: TODO: make sure that first executed action if the one with STARTED status (if there is already exactly one with this status)

File: src/ralph/lib/external_services/base.py [hunk 4]
```diff
@@ -50,3 +42,14 @@ def run(self, **kwargs):
         while job and not any([job.is_finished, job.is_failed]):
             time.sleep(0.1)
         return job.result
+
+    def run_async(self, **kwargs):
+        job = self.queue.enqueue(self.method, kwargs=kwargs)
+        return job
+
+
+class InternalService(ExternalService):
```
Comment (mkurek) [inline src/ralph/lib/external_services/base.py:51]: we should discuss if this is proper place for it (and if it should stay named `InternalService`)

File: .gitignore [hunk 5]
```diff
@@ -75,3 +75,5 @@ src/ralph/var/
 debian/ralph-core*
 
 .idea
+
+!src/ralph/lib
```
Comment (szok) [inline .gitignore:79]: :+1:

**Pull request closed without merging**